recipes/imake/all/test_package/conanfile.py

```diff
@@ -1,44 +1,40 @@
-from conans import AutoToolsBuildEnvironment, ConanFile, tools
-import contextlib
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.microsoft import is_msvc
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
+
 import os
 import shutil
 
-required_conan_version = ">=1.36.0"
+required_conan_version = ">=1.54.0"
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
-
     exports_sources = "Imakefile", "Imake.tmpl"
 
     def build_requirements(self):
-        if not tools.get_env("CONAN_MAKE_PROGRAM"):
-            self.build_requires("make/4.2.1")
-
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        if self.settings.compiler == "Visual Studio":
-            with tools.vcvars(self):
-                env = {
-                    "CC": "cl -nologo",
-                }
-                with tools.environment_append(env):
-                    yield
-        else:
-            yield
+        if not self.conf.get("tools.gnu:make_program", check_type=str):
+            self.tool_requires("make/4.2.1")
+
+    def generate(self):
+        deps = AutotoolsDeps(self)
+        deps.generate()
+
+        tc = AutotoolsToolchain(self)
+
+        env = tc.environment()
+        if is_msvc(self):
+            env.define("CC", "cl -nologo")
+        tc.generate(env)
 
     def build(self):
         for src in self.exports_sources:
             shutil.copy(os.path.join(self.source_folder, src), os.path.join(self.build_folder, src))
```

Comment:
```suggestion
            copy(self, "*", os.path.join(self.source_folder, src), os.path.join(self.build_folder, src)
```
Prefer Conan copy method, because it uses [robocoy](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/robocopy) which avoid some permission errors on Windows, which regular `shutil.copy` is not able to deal with.